### PR TITLE
[Highlight] Add user configurable timer

### LINKED ISSUE #187: [Highlight] Add configurable inactive timeout
# Feature request
#### Select the type of feature you are requesting:
- [x] Command

#### Describe your requested feature
Currently, the timeout value before a user is triggered for their highlight words is fixed.  We should add a per-user configurable timeout value, so that each user can choose how long they should be "inactive" for before they receive a message.  If the value is not set, it should fall back to the default.

### LINKED ISSUE #204: [Highlight] Add frequency filter for highlight
# Feature request

#### Select the type of feature you are requesting:

- [ ] Cog
- [x] Command
- [ ] API functionality

#### Describe your requested feature

Add a filter to prevent a user from getting spammed with the same highlight in the same channel repeatedly. If a highlight is brought up in conversation, it is easy for a user to get spammed with notifications. When a highlight is detected, have Ren check when the user last received a notification for that highlight and only notify them if the last notification was sent more than X seconds/minutes ago (X is some arbitrary amount of time, perhaps 5 minutes).

### LINKED ISSUE #205: [Highlight] Increase inactivity timer on renhighlight
# Feature request

#### Select the type of feature you are requesting:

- [ ] Cog
- [x] Command
- [ ] API functionality

#### Describe your requested feature

Please increase the inactivity timer check on highlight from 20 seconds, perhaps to 30 or 45 seconds.

### PR DESCRIPTION
**Pull request by Injabie3**
### Type
- [x] Enhancement

### Description of the changes
- Resolves #187.
- Resolves #204.
- Resolves #205.

This PR adds a per-user configurable timer to allow a user to:
1. Set the interval between consecutive triggers within the same channel.
2. Set how long they need to be inactive for to cause one of their highlight words to be triggered.

Currently, this time interval controls both of the aforementioned points.  A future PR could separate 2 from 1.

### CODE REVIEW

File: cogs/highlight.py [hunk 1]
```diff
@@ -310,6 +318,104 @@ def _registerUser(self, guildId, userId):
...
+            self._registerUser(guildId, userId)
+            self.highlights[guildId][userId][KEY_TIMEOUT] = seconds
+
+            confMsg = await self.bot.say("Timeout set to {} seconds.".format(seconds))
+            await self.settings.put(KEY_GUILDS, self.highlights)
+            await self.bot.delete_message(ctx.message)
+        await self._sleepThenDelete(confMsg, 5)
+
+
+    def _triggeredRecently(self, msg, uid, timeout=DEFAULT_TIMEOUT):
```
Comment (KamiCreed) [inline cogs/highlight.py:353]: Is underscores a usual convention for helper functions?

Comment (Injabie3) [reply cogs/highlight.py:353]: Typically used to indicate that it shouldn't be invoked directly from outside the class (e.g. internal use).

**Review by KamiCreed — APPROVED** [review]

File: cogs/highlight.py (hunk 1 above)
Comment (KamiCreed) [reply cogs/highlight.py:353]: Does python actually see that as like a private function?

Comment (Injabie3) [reply cogs/highlight.py:353]: Explicitly, no: you can still call it from outside, but static checkers won't like it.

**Pull request merged**